lib/portlayer/network/context.go

```diff
@@ -176,22 +219,30 @@ func reserveGateway(gateway net.IP, subnet *net.IPNet, ipam *IPAM) (net.IP, erro
 	return nil, fmt.Errorf("could not reserve gateway address for network %s", subnet)
 }
 
-func (c *Context) newScopeCommon(id uid.UID, name, scopeType string, subnet *net.IPNet, gateway net.IP, dns []net.IP, ipam *IPAM, network object.NetworkReference) (*Scope, error) {
+func (c *Context) addScope(s *Scope) error {
 	defer trace.End(trace.Begin(""))
+
+	if _, ok := c.scopes[s.name]; ok {
+		return DuplicateResourceError{}
+	}
+
 	var err error
-	var space *AddressSpace
 	var defaultPool bool
 	var allzeros, allones net.IP
+	var space *AddressSpace
+	spaces := s.spaces
```

Comment:
Yes I moved this code over from another function, just didn't want to mess with it too much.